src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -204,32 +251,475 @@ TEST_F(AlertRequestTest, Run_MandatoryParametersAreInvalid_UNSUCCESS) {
+  (*msg_)[am::strings::msg_params][am::strings::play_tone] = true;
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(mock_message_helper_,
+              ProcessSoftButtons((*msg_)[am::strings::msg_params], _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(_))
+      .Times(AtLeast(1))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::INVALID_ENUM;
+  Event ui_event(hmi_apis::FunctionID::UI_Alert);
+  ui_event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::TTS_StopSpeaking)))
+      .WillOnce(Return(true));
+
+  command->on_event(ui_event);
+
+  Event tts_stop_event(hmi_apis::FunctionID::TTS_StopSpeaking);
+  tts_stop_event.set_smart_object(*msg_);
+  command->on_event(tts_stop_event);
+
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::ABORTED;
+  Event event(hmi_apis::FunctionID::TTS_Speak);
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::ABORTED),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->on_event(event);
+}
+
+TEST_F(AlertRequestTest, OnEvent_TTSWarningUiAlertWarning_SUCCESS) {
+  AddAlertTextsToMsg();
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::WARNINGS;
+  (*msg_)[am::strings::msg_params][am::strings::play_tone] = true;
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(mock_message_helper_,
+              ProcessSoftButtons((*msg_)[am::strings::msg_params], _, _, _))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(_))
+      .Times(AtLeast(1))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  Event ui_event(hmi_apis::FunctionID::UI_Alert);
+  ui_event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::TTS_StopSpeaking)))
+      .WillOnce(Return(true));
+
+  command->on_event(ui_event);
+
+  Event tts_stop_event(hmi_apis::FunctionID::TTS_StopSpeaking);
+  tts_stop_event.set_smart_object(*msg_);
+  command->on_event(tts_stop_event);
+
+  Event event(hmi_apis::FunctionID::TTS_Speak);
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::WARNINGS),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->on_event(event);
+}
```

Comment:
@okozlovlux Here also a lot of code repetition in the last 3 tests. Please move the common part into a function.